tests/integrationtest/t/executor/insert.test

```diff
@@ -1620,3 +1620,25 @@ insert into parent values (2, 3) on duplicate key update ref = 3;
 --enable_warnings
 insert ignore into parent values (2, 3) on duplicate key update ref = 3;
 --disable_warnings
+
+# TestIssue55457
+drop table if exists t1, t2;
+create table t1 (id int primary key, col1 varchar(10) not null default '');
+create table t2 (id int primary key, col1 varchar(10));
+insert into t2 values (1, null);
+insert ignore into t1 values(5, null);
+set session sql_mode = '';
+-- error 1048
+insert into t1 values(1, null);
+-- error 1048
+insert into t1 set id = 1, col1 = null;
+-- error 1048
+insert t1 VALUES (5, 5) ON DUPLICATE KEY UPDATE col1 = null;
```

Comment:
What about inserting multiple values with `on duplicate key update`? Please add it.

Reply:
Updated in https://github.com/pingcap/tidb/pull/55477/commits/cc158dd04509cbb77720bca12c96054fc875fb50